### DISCUSSION TITLE
How can i implement rate-limiting to a Server Component/Action

### DISCUSSION THREAD

**Discussion by rahulps1000** [vercel/next.js#67775] [Help]
### Summary

I am creating a page for applying coupon code. the coupon code is passes in the url as a param and i want to prevent users from brute forcing the page. the page is a server component and hence no api call is done.

**Comment by icyJoseph** [comment]
I guess, you can use the `headers` function, to collect an IP/ or other identifier, from the incoming request, and rate limit that? IIRC, some 3rd party integrations have that built-in.
